src/uart/mod.rs

```diff
@@ -0,0 +1,73 @@
+//! Using the UART
+use crate::error::{NegativeErrorExt, NumericError};
+use embedded_io;
+use embedded_io::ErrorKind;
+use riot_sys::libc::c_void;
+use riot_sys::uart_init;
+use riot_sys::uart_t;
+use riot_sys::uart_write;
+
+#[derive(Debug)]
+pub struct UARTDevice {
+    dev: uart_t,
+}
+
+impl embedded_io::Error for NumericError {
+    fn kind(&self) -> ErrorKind {
+        match -self.number() as _ {
+            riot_sys::ENODEV => ErrorKind::NotFound,
+            riot_sys::ENOTSUP => ErrorKind::Unsupported,
+            _ => ErrorKind::Other,
+        }
+    }
+}
+
+impl embedded_io::ErrorType for UARTDevice {
+    type Error = NumericError;
+}
+
+impl UARTDevice {
+    /// Create a new UARTDevice from a RIOT descriptor
+    pub fn from_c(dev: uart_t) -> Self {
+        UARTDevice { dev }
+    }
+
+    /// Create a new UARTDevice from device number
+    pub fn from_port(dev: u8) -> Self {
+        UARTDevice { dev }
+    }
+
+    pub fn init(
```

Comment:
Anticipating that at some point we may want to have a means to use this with blocking or async reads, it may make sense to call this `init_with_callback`.

(I have some ideas on how `init_for_read` might work, and those will be implemented on some different T than what will eventually be part of the UARTDevice type.)